### PR TITLE
add support for caching flickr_api library results

### PR DESCRIPTION
**Pull request by mtrovo**
Add support for caching as the upstream library flickr_api supports it.

### CODE REVIEW

File: flickr_download/flick_download.py [hunk 1]
```diff
@@ -368,16 +391,35 @@ def main():
...
                         help='Save photo info like description and tags, one .json file per photo')
+    parser.add_argument('-c', '--cache', type=str, metavar='CACHE',
+                        help='Cache results on provided file')
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="increase output verbosity")
     parser.set_defaults(**_load_defaults())
 
     args = parser.parse_args()
 
+    if args.verbose:
```
Comment (beaufour) [inline flickr_download/flick_download.py]: You snuck in this one, but I like it :)

File: flickr_download/flick_download.py [hunk 2]
```diff
@@ -368,16 +391,35 @@ def main():
...
+        logging.getLogger().setLevel(logging.DEBUG)
+
+    cache = None
+    if args.cache:
+        cache = get_cache(args.cache)
+        Flickr.enable_cache(cache)
+
+        def signal_handler(signal, frame):
+            save_cache(args.cache, cache)
+            sys.exit(0)
```
Comment (beaufour) [inline flickr_download/flick_download.py]: Don’t we want to exit with a return code, so that the caller can act accordingly? (Or is my signal handler knowledge just really outdated?)

File: flickr_download/flick_download.py [hunk 3]
```diff
@@ -368,16 +391,35 @@ def main():
...
+
+        print('Cache is enabled')
+
     if args.list_naming:
         print(get_filename_handler_help())
         return 1
 
     if not args.api_key or not args.api_secret:
-        print ('You need to pass in both "api_key" and "api_secret" arguments', file=sys.stderr)
+        print('You need to pass in both "api_key" and "api_secret" arguments', file=sys.stderr)
```
Comment (beaufour) [inline flickr_download/flick_download.py]: Why this change? It doesn’t follow the style of the rest of the file?

File: flickr_download/flick_download.py [hunk 4]
```diff
@@ -300,6 +303,26 @@ def print_sets(username):
         print('{0} - {1}'.format(photo.id, photo.title))
 
 
+def get_cache(path):
+    if not os.path.exists(path):
+        cache = SimpleCache(max_entries=20000, timeout=3600)
```
Comment (beaufour) [inline flickr_download/flick_download.py]: Complete nit, but I’m a huge fans of early returns. So if you just do a `return SimpleCache(...` here you can lose the `else` below and the indent. Makes it easier to read imo.

File: flickr_download/flick_download.py (hunk 2 above)
Comment (mtrovo) [reply flickr_download/flick_download.py]: good catch, I'm changing it

File: flickr_download/flick_download.py (hunk 3 above)
Comment (mtrovo) [reply flickr_download/flick_download.py]: pyflakes complained about this one, but I can revert

Comment (beaufour) [reply flickr_download/flick_download.py]: Weird, I get no errors from `flake8` on it

**Comment by beaufour** [comment]
As per #48 there are at least one change that will have to be fixed because of the new API before I can merge. It automatically sets the file name extension. Doesn’t need to be part of your PR at all :), just saying that the new API needs some more testing (and potential fixing) in general
